src/main/Java/SampleApp.java

```diff
@@ -0,0 +1,267 @@
+	private static final String DEFAULT_PROPERTIES_FILE = "environment.properties";
+	private static final Logger logger = LogManager.getLogger(SampleApp.class);
+	private static Properties props;
+	
+	/**
+	 * This application makes sample API calls to the CHPL API:
+	 * - shows how to make API calls using org.apache.http
+	 * - shows how to parse the JSON response using com.google.gson.jsonparser
+	 */
+	public SampleApp(){}
+	
+	public static void main(String[] args) throws IOException, URISyntaxException, JSONException {
+		SampleApp sampleApp = new SampleApp();
+		String token = null;
+		
+		props = sampleApp.loadProperties(DEFAULT_PROPERTIES_FILE);
+		sampleApp.displayInstructions();
+		Integer input = sampleApp.getValidUserInput();
+		
+		/**
+		 * Example 1:
+		 * a. Call the API /auth/authentication using HTTP POST to get a valid token (note: the authentication token is not necessary for many API calls, including /search)
+		 * b. Parse JSON response using com.google.gson
+		 * c. Return token as string
+		 */
+		if(input == 0 || input == 1 || input == 4){
+			// This token could be used in an API call that requires authentication. 
+			token = sampleApp.getToken();
+		}
+		
+		/**
+		 * Example 2: 
+		 * a. Call the API /search using HTTP GET to get all Certified Product Details
+		 * b. Parse the JSON using org.json to get only information keyed on "product"
+		 * c. Parse the "product" JSON to CSV format
+		 * d. Write the csv data to a file
+		 */
+		if(input == 0 || input == 2){
+			String searchResult = sampleApp.getSearchResult();
+			JSONObject search = new JSONObject(searchResult);
+			JSONArray searchArr = search.getJSONArray("results");
+			JSONObject products = new JSONObject();
+			
+			for (Integer i = 0; i < searchArr.length(); i++) {
+			      if(searchArr.optJSONObject(i).has("product")){
+			    	  products.put(searchArr.optJSONObject(i).get("id").toString(), searchArr.optJSONObject(i).get("product"));
+			      }
+			 }
+			
+			System.out.println("Parsed search JSONArray to retrieve products as follows: \n" + products.toString());
+			JSONArray productNames = products.toJSONArray(products.names());
+			System.out.println("Parsed certified products JSON to get products as follows: \n" + productNames.toString());
+			String csv = CDL.toString(productNames);
+			System.out.println("Parsed products JSON to CSV as follows: \n" + csv);
+			File file = new File("products.csv");
+			FileUtils.writeStringToFile(file, csv);
+			System.out.println("Saved product information to products.csv");
+		}
+		
+		/**
+		 * Example 3: 
+		 * a. call the API /search to get all Certified Product Details
+		 * b. Save results to a .txt file
+		 */
+		if(input == 3 || input == 0){
+			sampleApp.getSearchResult_saveResultsInFile();
+		}
+		
+		/**
+		 * Example 4:
+		 * a. Call the API /data/search_options using token for authentication
+		 */
+		if(input == 4 || input == 0){
+			sampleApp.getSearchOptions_usesAuthentication(token);
```

Comment:
No; is it okay if I add a comment that authentication is not required for this API call, and that this is just an example of how to add authorization?